app/controllers/carto/api/multifactor_auths_controller.rb

```diff
@@ -0,0 +1,69 @@
+class Carto::Api::MultifactorAuthsController < ::Api::ApplicationController
+  include Carto::ControllerHelper
+
+  ssl_required
+
+  before_action :load_user
+  before_action :check_ff
+  before_action :check_shared_secret_not_present, only: [:create, :verify_code]
+  before_action :load_multifactor_auth, only: [:show, :verify_code, :destroy]
+
+  rescue_from Carto::UnprocesableEntityError, with: :rescue_from_carto_error
+  rescue_from Carto::UnauthorizedError, with: :rescue_from_carto_error
+
+  def create
+    multifactor_auth = @carto_viewer.user_multifactor_auths.create!(create_params)
+    render_jsonp(Carto::Api::MultifactorAuthPresenter.new(multifactor_auth).to_poro_with_qrcode, 201)
+  rescue ActionController::ParameterMissing => e
+    raise Carto::UnprocesableEntityError.new(e.message)
+  rescue ActiveRecord::RecordInvalid => e
+    raise Carto::UnprocesableEntityError.new(e.message)
+  end
+
+  def verify_code
+    @multifactor_auth.verify!(params[:code])
+    render_jsonp(Carto::Api::MultifactorAuthPresenter.new(@multifactor_auth).to_poro, 200)
+  rescue ActiveRecord::RecordInvalid => e
+    raise Carto::UnprocesableEntityError.new(e.message)
+  end
+
+  def destroy
+    @multifactor_auth.destroy
+    render_jsonp(Carto::Api::MultifactorAuthPresenter.new(@multifactor_auth).to_poro, 200)
+  end
+
+  def show
+    render_jsonp(Carto::Api::MultifactorAuthPresenter.new(@multifactor_auth).to_poro, 200)
+  end
+
+  def index
+    auths = @carto_viewer.user_multifactor_auths.map do |auth|
+      Carto::Api::MultifactorAuthPresenter.new(auth).to_poro
+    end
+
+    render_jsonp(auths, 200)
```

Comment:
Why `jsonp`? I think we can make it a straight JSON API

```suggestion
    render json: auths, status: 200
```

Same in other places of the controller. And same comment about not investing too much into this if we are going to change it later.

Reply:
Yep you are right, I mostly `copied` the api_key API and didn't think too much on the usage.